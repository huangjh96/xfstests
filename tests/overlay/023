#! /bin/bash
# FS QA Test 023
#
# The "work" directory in an overlyfs workdir should not inherit
# ACL attrs from workdir.
# Kernel commits below fix it.
#   c11b9fdd6a61 ovl: remove posix_acl_default from workdir
#   e1ff3dd1ae52 ovl: fix workdir creation
#
# This reproducer was originally written by
#     Miklos Szeredi <mszeredi@redhat.com>
#
#-----------------------------------------------------------------------
# Copyright (c) 2017 Red Hat Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/attr

# remove previous $seqres.full before test
rm -f $seqres.full

# real QA test starts here

# Modify as appropriate.
_supported_fs overlay
_supported_os Linux
_require_acls
_require_scratch

# Remove all files from previous tests
_scratch_mkfs

# setting acls before mount
wkdir=$OVL_BASE_SCRATCH_MNT/$OVL_WORK
mkdir -p $wkdir
setfacl -d -m o::rwx $wkdir

_scratch_mount

# getting acls, ACL set on workdir should be cleaned
getfacl -p $wkdir/work 2>&1 | _filter_scratch | _filter_ovl_dirs

# success, all done
status=0
exit
