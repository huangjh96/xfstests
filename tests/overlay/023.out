QA output created by 023
# file: SCRATCH_DEV/OVL_WORK/work
# owner: root
# group: root
user::---
group::---
other::---

